berkdb/rep/rep_record.c

```diff
@@ -753,15 +819,25 @@ send:			if (__rep_send_message(dbenv,
 	case REP_FILE_REQ:
 		MASTER_ONLY(rep, rp);
 		ret = __rep_send_file(dbenv, rec, *eidp);
+		fromline = __LINE__;
 		goto errlock;
 #endif
 	case REP_LOG:
 	case REP_LOG_MORE:
 		CLIENT_ONLY(rep, rp);
 		MASTER_CHECK(dbenv, *eidp, rep);
-		if ((ret =
-			__rep_apply(dbenv, rp, rec, ret_lsnp, commit_gen)) != 0)
+		if (!IN_ELECTION_TALLY(rep)) {
```

Comment:
Prevent a replicant from processing log-messages if it is an election candidate.

Reply:
This closes a race that Kyle pointed out:
"... if you allow log messages to hit candidates, then I think you can wind up in a situation like... n1, the old primary, tries to repllcate write w1. n2, isolated from n1, declares candidacy for leadership, without w1. It asks n3, n4 to vote for it, and neither of them have w1, and they grant their votes to n2. w1 arrives on n3 and n5, and n1 now knows w1 is committed. n2 becomes master, and n3 rolls *back* w1."